### ISSUE TITLE
WPF: Zoom is laggy after MapControl is re-created 

### ISSUE THREAD

**Issue by EmaDozer** [Mapsui/Mapsui#2700]
https://github.com/Mapsui/Mapsui/assets/93271991/bf2cadb8-1851-4e36-b0f9-f45a054f604b

**Mapsui Version**
4.1.7

**Mapsui Platform**
WPF

**Device**
Windows

**Describe the bug**
I discovered a weird behavior where the zoom is laggy if the MapControl is in a TabItem of a TabConrol and the tabs are dynamically created. This was no problem in Mapsui version 3.

The reason is the re-creation of the MapControl if the tab is re-selected (dynamic creation of tabs). Its the normal behavior of the internal virtualization of TabControl. In addition the state of the Map property is saved and is always assigned as the Map of the MapControl as soon as it gets re-created. If the Map property is always initialized new, the zoom is smooth. The map only is re-created, if a tab with another content (ex TextBlock) also exists.

**To Reproduce**
To see the effect, switch the tab to the TextBlock and back to MapControl. Sometimes it needs a couple of switches. It seems, that the lag gets worse with each switch/re-creation of the MapControl. Don’t know why. 

I attached a sample project
[MapsuiTab.zip](https://github.com/user-attachments/files/16068519/MapsuiTab.zip)

https://github.com/Mapsui/Mapsui/assets/93271991/67cb06a7-4b9d-4f07-86ee-f545df222fe0

**Comment by pauldendulk** [comment]
Thanks for your excellent bug report. I can reproduce the behavior with your sample. The problem also exists in 5.0.0-beta.1.

**Comment by pauldendulk** [comment]
Inspecting the code I see there is a CustomMap derived from MapControl which is IDisposable, so CustomMap must be disposed. How should that work in this setup where a standard TabControl is used with content defined in xaml?

**Comment by EmaDozer** [comment]
You are correct, the MapControl is not disposed. Do you think the laggy zoom is related to this?

**Comment by EmaDozer** [comment]
Indeed, i managed to dispose MapControl via an AttachedProperty as soon as it is released from the TabControl and this seems to solve the issue. 

I dont really understand why, do you have an explanation? As i mentioned above, this was no Issue in v3.
